lib/ansible/modules/network/meraki/meraki_content_filtering_facts.py

```diff
@@ -0,0 +1,170 @@
+    org_id:
+        description:
+        - ID of organization associated to a network.
+    subset:
+        description:
+        - Display only certain facts
+        choices: [categories, policy]
+
+author:
+    - Kevin Breit (@kbreit)
+extends_documentation_fragment: meraki
+'''
+
+EXAMPLES = r'''
+- name: List all content filtering information
+  meraki_content_filtering_facts:
+    auth_key: abc12345
+    org_name: YourOrg
+    net_name: YourNet
+  delegate_to: localhost
+
+- name: List content filtering policy information
+  meraki_content_filtering_facts:
+    auth_key: abc12345
+    org_name: YourOrg
+    net_name: YourNet
+    subset: policy
+  delegate_to: localhost
+
+- name: List all content filtering categories
+  meraki_content_filtering_facts:
+    auth_key: abc12345
+    org_name: YourOrg
+    net_name: YourNet
+    subset: categories
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+data:
+    description: Information about the created or manipulated object.
+    returned: info
+    type: complex
+    contains:
+      id:
+        description: Identification string of network.
+        returned: success
+        type: str
+        sample: N_12345
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule, json, env_fallback
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_native
+from ansible.module_utils.network.meraki.meraki import MerakiModule, meraki_argument_spec
+
+
+def main():
+
+    # define the available arguments/parameters that a user can pass to
+    # the module
+
+    argument_spec = meraki_argument_spec()
+    argument_spec.update(
+        net_id=dict(type='str'),
+        net_name=dict(type='str', aliases=['network']),
+        subset=dict(type='str', choices=['categories', 'policy'])
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False,
+                           )
+
+    meraki = MerakiModule(module, function='content_filtering_facts')
+    module.params['follow_redirects'] = 'all'
+
+    category_urls = {'content_filtering_facts': '/networks/{net_id}/contentFiltering/categories'}
+    policy_urls = {'content_filtering_facts': '/networks/{net_id}/contentFiltering'}
+
+    meraki.url_catalog['categories'] = category_urls
+    meraki.url_catalog['policy'] = policy_urls
+
+    if meraki.params['net_name'] and meraki.params['net_id']:
+        meraki.fail_json(msg='net_name and net_id are mutually exclusive')
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
```

Comment:
The check-mode test should come as late as possible. There are various failure-cases that you are skipping in check-mode.

Reply:
I don’t use check_mode right now since Meraki doesn’t support it. I’ll move it down, but the code doesn’t serve much purpose except to act as a placeholder for the future if it’s added.